### DISCUSSION TITLE
Airflow Ignore for negation folders

### DISCUSSION THREAD

**Discussion by panchaml** [apache/airflow#37773] [Q&A]
I have my repository as
```cdm_dags/
    - .airflowignore
    - requirements.txt
    - scheduler.py
    - config/
    - dev/
        - engine/
            - dags/
            - src_code/
            - config/
        - common/
            - dags/
            - config/
            - res/
            - something.yaml
        - code_dir/
            - dags/
            - config/
            - code/
            - something_else.yaml
```

I have added airflowignore file in the root directory, but I want it to only check the dags directory under each directory (engine, common, code_dir)

I have tried specifying all the respective directories to be ignored like:
```
dev/engine/src_code/
dev/engine/config/

dev/common/config/
dev/common/res/
dev/common/something.yaml

dev/code_dir/config/
dev/code_dir/code/
dev/code_dir/something_else.yaml
```
This isn't the best method, I tried using negation to specify like:
```
# ignore all files/sub-dirs from root dir
*

# whitelist engine dir dags
!/dev/engine/dags

# whitelist common dir dags
!/dev/common/dags

# whitelist code_dir dir dags
!/dev/code_dir/dags

```
But this also didn't work. 
Do provide an optimal solution.
PS. I also want to ignore all the files in the dev dir also.

**Comment by potiuk** [comment]
Airlfow does not support negation. You need to figure out the way that fits best to you - you should use regular expressions or glob:

https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dags.html#airflowignore

One suggestion to explore for you is to add regexp that will match all .py files in directories you want to ignore

Ideally refactor your structure of folders to better fit your case. Also - if you think negation might be useful - you migh contribute such feature (sounds like a good idea).
